lasagne/layers/recurrent.py

```diff
@@ -1,55 +1,98 @@
+"""
+Layers to construct recurrent networks. Recurrent layers can be used similarly
+to feed-forward layers except that the input shape is expected to be
+(batch_size, sequence_length, num_inputs).
+
+The following recurrent layers are implemented:
+
+* :func:`CustomRecurrentLayer()`
+* :func:`RecurrentLayer()`
+* :func:`LSTMLayer()`
+
+Recurrent layers and feed-forward layers can be combined in the same network
+by using a few reshape operations, please refer to the recurrent examples for
+further explanations.
+
+"""
 import theano
 import theano.tensor as T
 from .. import nonlinearities
 from .. import init
-from .. import utils
 
 from .base import Layer
 from .input import InputLayer
 from .dense import DenseLayer
 from . import helper
 
+__all__ = [
+    "CustomRecurrentLayer",
+    "RecurrentLayer",
+    "LSTMLayer"
+]
+
 
 class CustomRecurrentLayer(Layer):
-    '''
+    """
     A layer which implements a recurrent connection.
 
     Expects inputs of shape
         (n_batch, n_time_steps, n_features_1, n_features_2, ...)
-    '''
-    def __init__(self, input_layer, input_to_hidden, hidden_to_hidden,
+
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape
+    input_to_hidden : nntools.layers.Layer
+        Layer which connects input to the hidden state
+    hidden_to_hidden : nntools.layers.Layer
+        Layer which connects the previous hidden state to the new state
+    nonlinearity : function or theano.tensor.elemwise.Elemwise
+        Nonlinearity to apply when computing new state
+    hid_init : function or np.ndarray or theano.shared
+        Initial hidden state
+    backwards : boolean
+        If True, process the sequence backwards
+    learn_init : boolean
+        If True, initial hidden values are learned
+    gradient_steps : int
+        Number of timesteps to include in backpropagated gradient
+        If -1, backpropagate through the entire sequence
+    return_sequence : boolean
+        Specifies if the LSTM should output the hidden state for all positions
+        in the sequence or only for the last position. If true the output
+        shape is (num_batch, sequence_lengt, num_units) if false the output
+        shape is (num_batch, num_units).
+    grad_clipping: False or float
+        If float the gradient messages are clipped during the backward pass.
+        See [2]_ (p. 6) for further explanation.
```

Comment:
I think this reference ([2]_) will be undefined because you define it in the `LSTMLayer` docstring.